### PR TITLE
Mappings for net/minecraft/world/level/levelgen

### PR DESCRIPTION
**Pull request by PiTheGuy**

### CODE REVIEW

File: data/net/minecraft/world/level/levelgen/structure/structures/WoodlandMansionPieces.mapping [hunk 1]
```diff
@@ -12,6 +12,9 @@ CLASS net/minecraft/world/level/levelgen/structure/structures/WoodlandMansionPie
 			ARG 1 width
 			ARG 2 height
 			ARG 3 valueIfOutside
+		METHOD edgesTo (III)Z
+			ARG 1 x
+			ARG 2 y
```
Comment (Lanse505) [inline data/net/minecraft/world/level/levelgen/structure/structures/WoodlandMansionPieces.mapping:17]: ARG 3 expectedValue

**Review by Lanse505 — DISMISSED** [review]

File: data/net/minecraft/world/level/levelgen/feature/rootplacers/RootPlacer.mapping [hunk 2]
```diff
@@ -15,11 +15,16 @@ CLASS net/minecraft/world/level/levelgen/feature/rootplacers/RootPlacer
...
 		ARG 5 treeConfig
 	METHOD placeRoots (Lnet/minecraft/world/level/LevelSimulatedReader;Ljava/util/function/BiConsumer;Lnet/minecraft/util/RandomSource;Lnet/minecraft/core/BlockPos;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/levelgen/feature/configurations/TreeConfiguration;)Z
 		ARG 1 level
 		ARG 2 blockSetter
 		ARG 3 random
+		ARG 4 pos
+		ARG 5 trunkOrigin
 		ARG 6 treeConfig
+	METHOD rootPlacerParts (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/Products$P3;
+		ARG 0 codecBuilder
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/feature/rootplacers/RootPlacer.mapping]:
```suggestion
		ARG 0 instance
```

N.B. A brief [search](https://github.com/search?q=repo%3AParchmentMC%2FParchment%20RecordCodecBuilder%24Instance&type=code) shows we name this all sorts of things (most commonly `instance`, but also `in`, `locations`, `codec` among others), I think `instance` is the best + most commonly accepted name for this particular pattern.

File: data/net/minecraft/world/level/levelgen/feature/stateproviders/NoiseBasedStateProvider.mapping [hunk 3]
```diff
@@ -6,3 +6,5 @@ CLASS net/minecraft/world/level/levelgen/feature/stateproviders/NoiseBasedStateP
 	METHOD getNoiseValue (Lnet/minecraft/core/BlockPos;D)D
 		ARG 1 pos
 		ARG 2 delta
+	METHOD noiseCodec (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/Products$P3;
+		ARG 0 codecBuilder
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/feature/stateproviders/NoiseBasedStateProvider.mapping]:
```suggestion
		ARG 0 instance
```

File: data/net/minecraft/world/level/levelgen/feature/stateproviders/NoiseProvider.mapping [hunk 4]
```diff
@@ -11,3 +11,5 @@ CLASS net/minecraft/world/level/levelgen/feature/stateproviders/NoiseProvider
 		ARG 1 possibleStates
 		ARG 2 pos
 		ARG 3 delta
+	METHOD noiseProviderCodec (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/Products$P4;
+		ARG 0 codecBuilder
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/feature/stateproviders/NoiseProvider.mapping]:
```suggestion
		ARG 0 instance
```

File: data/net/minecraft/world/level/levelgen/structure/Structure.mapping [hunk 5]
```diff
@@ -47,12 +48,16 @@ CLASS net/minecraft/world/level/levelgen/structure/Structure
...
 		ARG 0 stub
 		ARG 1 context
+	METHOD lambda$findValidGenerationPoint$2 (Lnet/minecraft/world/level/levelgen/structure/Structure$GenerationContext;Lnet/minecraft/world/level/levelgen/structure/Structure$GenerationStub;)Z
+		ARG 1 stub
 	METHOD onTopOfChunkCenter (Lnet/minecraft/world/level/levelgen/structure/Structure$GenerationContext;Lnet/minecraft/world/level/levelgen/Heightmap$Types;Ljava/util/function/Consumer;)Ljava/util/Optional;
 		ARG 0 context
 		ARG 1 heightmapTypes
 		ARG 2 generator
 	METHOD settingsCodec (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/serialization/codecs/RecordCodecBuilder;
 		ARG 0 codecBuilder
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/structure/Structure.mapping]:
```suggestion
		ARG 0 instance
```

File: data/net/minecraft/world/level/levelgen/structure/placement/ConcentricRingsStructurePlacement.mapping [hunk 6]
```diff
@@ -14,3 +14,5 @@ CLASS net/minecraft/world/level/levelgen/structure/placement/ConcentricRingsStru
 		ARG 7 spread
 		ARG 8 count
 		ARG 9 preferredBiomes
+	METHOD codec (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/Products$P9;
+		ARG 0 codecBuilder
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/structure/placement/ConcentricRingsStructurePlacement.mapping]:
```suggestion
		ARG 0 instance
```

File: data/net/minecraft/world/level/levelgen/structure/placement/StructurePlacement.mapping [hunk 7]
```diff
@@ -20,11 +20,18 @@ CLASS net/minecraft/world/level/levelgen/structure/placement/StructurePlacement
...
+		ARG 3 regionX
+		ARG 4 regionZ
+		ARG 5 probability
 	METHOD legacyProbabilityReducerWithDouble (JIIIF)Z
 		ARG 0 baseSeed
 		ARG 3 chunkX
 		ARG 4 chunkZ
 		ARG 5 probability
+	METHOD placementCodec (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/Products$P5;
+		ARG 0 codecBuilder
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/structure/placement/StructurePlacement.mapping]:
```suggestion
		ARG 0 instance
```

**Review by alcatrazEscapee — CHANGES_REQUESTED** [review]

**Review by alcatrazEscapee — APPROVED** [review]

**Pull request merged**